src/main/java/com/questhelper/quests/holygrail/HolyGrail.java

```diff
@@ -0,0 +1,362 @@
+		findFisherKing.addStep(inGalahadHouse, talkToGalahad);
+
+		findFisherKing.setLockingCondition(hasTwoWhistles);
+
+		steps.put(4, findFisherKing);
+
+		ConditionalStep findPercival = new ConditionalStep(this, talkToKingArthur2);
+		findPercival.addStep(hasFeather, openSack);
+
+		steps.put(8, findPercival);
+
+		ConditionalStep finishQuest = new ConditionalStep(this, goToTeleportLocation2);
+		finishQuest.addStep(hasGrail, talkToKingArthur3);
+		finishQuest.addStep(inFisherKingCastle2ThirdFloor, takeGrail);
+		finishQuest.addStep(inFisherKingCastle2SecondFloor, goUpNewCastleLadder);
+		finishQuest.addStep(inFisherKingCastle2BottomFloor, goUpNewCastleStairs);
+		finishQuest.addStep(inFisherKingRealm, openFisherKingCastleDoor);
+		finishQuest.addStep(inTeleportLocation, blowWhistle2);
+
+		steps.put(9, finishQuest);
+
+		return steps;
+	}
+
+	public void setupItemRequirements()
+	{
+		excalibur = new ItemRequirement("Excalibur", ItemID.EXCALIBUR);
+		holyTableNapkin = new ItemRequirement("Holy Table Napkin", ItemID.HOLY_TABLE_NAPKIN);
+		twoMagicWhistles = new ItemRequirement("Magic Whistles", ItemID.MAGIC_WHISTLE, 2);
+		threeCamelotTele = new ItemRequirement("Camelot Teleports", ItemID.CAMELOT_TELEPORT, 3);
+		ardyTele = new ItemRequirement("Ardougne Teleport", ItemID.ARDOUGNE_TELEPORT);
+		faladorTele = new ItemRequirement("Falador Teleport", ItemID.FALADOR_TELEPORT);
+		sixtyCoins = new ItemRequirement("Coins", ItemID.COINS_995, 60);
+		antipoison = new ItemRequirement("Antipoison", ItemID.ANTIPOISON4);
+		food = new ItemRequirement("Food", -1, -1);
+		combatGear = new ItemRequirement("A weapon and armour", -1, -1);
+		emptyInvSpot = new ItemRequirement("Empty Inventory Spot", -1, 1);
+		goldFeather = new ItemRequirement("Magic gold feather", ItemID.MAGIC_GOLD_FEATHER);
+		grailBell = new ItemRequirement("Grail Bell", ItemID.GRAIL_BELL);
+		oneMagicWhistle = new ItemRequirement("Magic Whistle", ItemID.MAGIC_WHISTLE);
+		grail = new ItemRequirement("Holy Grail", ItemID.HOLY_GRAIL);
+
+		highlightMagicWhistle1 = new ItemRequirement("Magic Whistle", ItemID.MAGIC_WHISTLE, 2);
+		highlightMagicWhistle1.setHighlightInInventory(true);
+
+		highlightMagicWhistle2 = new ItemRequirement("Magic Whistle", ItemID.MAGIC_WHISTLE);
+		highlightMagicWhistle2.setHighlightInInventory(true);
+
+		highlightGrailBell = new ItemRequirement("Grail Bell", ItemID.GRAIL_BELL);
+		highlightGrailBell.setHighlightInInventory(true);
+	}
+
+	public void loadZones()
+	{
+		camelotGround = new Zone(new WorldPoint(2744, 3517, 0), new WorldPoint(2733, 3483, 0));
+		camelotUpstairsZone1 = new Zone(new WorldPoint(2768, 3517, 1), new WorldPoint(2757, 3506, 1)); //og. 2764, 3517. 2748, 3501
+		camelotUpstairsZone2 = new Zone(new WorldPoint(2764, 3517, 1), new WorldPoint(2748, 3496, 1));
+		merlinRoom = new Zone(new WorldPoint(2768, 3505, 1), new WorldPoint(2765, 3496, 1));
+		entranaBoat = new Zone(new WorldPoint(2841,3332,1), new WorldPoint(2823,3328,1));
```

Comment:
I'd extend this + the entrana Island to range from planes 0-2. Although we don't expect a player to change plane, it's possible and an easy thing to catch with a single digit change.

Reply:
are you including camelotUpstraisZone1 and 2 in expanding the planes? Because that is meant to be a specific region on the same floor as Merlin.  Or are you specifically talking about expanding the planes of the entranaBoat and entranaIsland locations?

Reply:
Specifically the entrana ones.